packages/@lwc/engine-core/src/framework/freeze-template.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import {
+    ArrayPush,
+    ArraySplice,
+    ArrayUnshift,
+    ArrayCopyWithin,
+    ArrayFill,
+    ArrayPop,
+    ArrayShift,
+    ArraySort,
+    ArrayReverse,
+    defineProperty,
+    getOwnPropertyDescriptor,
+    isUndefined,
+} from '@lwc/shared';
+import { logError } from '../shared/logger';
+import { Template } from './template';
+import { TemplateStylesheetFactories } from './stylesheet';
+
+// See @lwc/engine-core/src/framework/template.ts
+const TEMPLATE_PROPS = ['slots', 'stylesheetToken', 'stylesheets', 'renderMode'] as const;
+
+// Via https://www.npmjs.com/package/object-observer
+const ARRAY_MUTATION_METHODS = [
```

Comment:
This bit seems like it should live in `@lwc/shared`. Possibly `getOriginalArrayMethod` also.

Reply:
I kind of feel it should rather live here, since I don't think any other code will want to import it. It makes it simpler to remove when the warning is replaced with `Object.freeze()`.

I suppose one could say the same thing about the new Array methods I added (e.g. `ArrayReverse`), but there was already a precedent for those things being in `@lwc/shared`.

Reply:
Hmm... I suppose that's fair, since this is a temporary fix. 👍